File: java/06_let_it_work/agent-infrastructure/src/main/java/com/workshop/agent/infrastructure/ConsoleDisplay.java
package com.workshop.agent.infrastructure;

import com.workshop.agent.application.Display;
import com.workshop.agent.application.Message;

public class ConsoleDisplay implements Display {
    @Override
    public void show(Message message) {
        String role = message.role();
        String capitalizedRole = Character.toUpperCase(role.charAt(0)) + role.substring(1);
        System.out.println(capitalizedRole + ": " + message.content());
    }
}

File: csharp/06_let_it_work/Agent/Agent.Infrastructure/ConsoleDisplay.cs
using Agent.Application;

namespace Agent.Infrastructure;

public class ConsoleDisplay : IDisplay
{
    public void Show(Message message)
    {
        var role = char.ToUpper(message.Role[0]) + message.Role.Substring(1);
        Console.WriteLine($"{role}: {message.Content}");
    }
}
